modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -460,7 +460,13 @@ def map_func(left, right=right, kwargs=kwargs):
                 return pandas.merge(left, right, **kwargs)
 
             new_self = self.__constructor__(
-                self._modin_frame.apply_full_axis(1, map_func)
+                self._modin_frame.apply_full_axis(
+                    axis=1,
+                    func=map_func,
+                    # We're going to explicitly change the shape across the 1-axis,
+                    # so we want for partitioning to adapt as well
+                    keep_partitioning=False,
```

Comment:
It seems that this flag was supposed to be used for the axis on which the `copartition` is called. But in these cases, this requirement applies to the other axis. Looks strange.

Reply:
What do you mean?

The `keep_partitioning` flag indicates that the number of splits along the passed axis has to be re-computed based on the resulting axis length. As in 'join' and 'merge' we're adding new columns to the frame the 1-axis will definitely grow, thus maybe requiring an updated number of splits.

P.S. 0-axis may also grow as a result of such operations, but we can't control its partitioning by any flag.

Reply:
Probably this parameter (`True` value) makes sense only if the shape of the dataframe does not change.